File: Assets/MLRepo/ml.72201f1e-ba18-487e-adc4-4381ae10e21b.davidmatos_doortrigger/DoorTrigger/Scripts/TriggerVolume.lua
Namespace("davidmatos_doortrigger");
do -- script TriggerVolume 
	
	-- get reference to the script
	local TriggerVolume = LUA.script;
	
    --|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||

	local doorObject = SerializedField("Door Object", GameObject);

    --|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||

    function TriggerVolume.Start()

    end

    function TriggerVolume.Update()

    end

    --NOTE: The local player is the only one that will trip these
    function TriggerVolume.OnTriggerEnter(other)
        if(other.gameObject.IsPlayer()) then
            local doorObjectScript = doorObject.GetComponent(davidmatos_doortrigger.DoorBase); -- get the script

            LuaEvents.InvokeLocalForAll(doorObjectScript, "OpenDoor"); --call the event on all clients
        end 
    end

    --NOTE: The local player is the only one that will trip these
    function TriggerVolume.OnTriggerExit(other)
        if(other.gameObject.IsPlayer()) then
            local doorObjectScript = doorObject.GetComponent(davidmatos_doortrigger.DoorBase); -- get the script

            LuaEvents.InvokeLocalForAll(doorObjectScript, "CloseDoor");--call the event on all clients
        end 
    end
end

File: Assets/MLRepo/ml.72201f1e-ba18-487e-adc4-4381ae10e21b.davidmatos_doortrigger/DoorTrigger/Scripts/TriggerVolume.cs
using ML.SDK;
using UnityEngine;

namespace DavidMatos.DoorTrigger
{
    public class TriggerVolume : MonoBehaviour
    {
        [Header("Door Reference")]
        [SerializeField] private DoorBase door;  // reference directly as DoorBase instead of GameObject

        private void OnTriggerEnter(Collider other)
        {
            if (door == null) return;

            MLPlayer player = other.gameObject.GetPlayer();
            if (player != null)
            {
                // local player is the only one who triggers this,
                // DoorBase will handle broadcasting if we're master
                door.OpenDoor();
            }
        }

        private void OnTriggerExit(Collider other)
        {
            if (door == null) return;

            MLPlayer player = other.gameObject.GetPlayer();
            if (player != null)
            {
                door.CloseDoor();
            }
        }
    }
}
